process/smartContract/builtInFunctions/esdtNFTCreateRoleTransfer.go

```diff
@@ -147,6 +159,50 @@ func (e *esdtNFTCreateRoleTransfer) executeTransferNFTCreateChangeAtCurrentOwner
 	return outAcc, nil
 }
 
+func (e *esdtNFTCreateRoleTransfer) deleteCreateRoleFromAccount(
+	acntDst state.UserAccountHandler,
+	esdtTokenRoleKey []byte,
+) error {
+	roles, _, err := getESDTRolesForAcnt(e.marshalizer, acntDst, esdtTokenRoleKey)
+	if err != nil {
+		return err
+	}
+
+	deleteRoles(roles, [][]byte{[]byte(core.ESDTRoleNFTCreate)})
+	return saveRolesToAccount(acntDst, esdtTokenRoleKey, roles, e.marshalizer)
+}
+
+func (e *esdtNFTCreateRoleTransfer) addCreateRoletoAccount(
+	acntDst state.UserAccountHandler,
+	esdtTokenRoleKey []byte,
+) error {
+	roles, _, err := getESDTRolesForAcnt(e.marshalizer, acntDst, esdtTokenRoleKey)
+	if err != nil {
+		return err
+	}
+
+	roles.Roles = append(roles.Roles, []byte(core.ESDTRoleNFTCreate))
```

Comment:
What happens if the role already exists, and it is added again?

Reply:
it should be impossible as role is unique at the metachain system sc level. Only if their is a bug in the protocol.